src/main/java/com/coreos/jetcd/cluster/MemberAddResponse.java

```diff
@@ -34,7 +30,17 @@ public Member getMember() {
    * returns a list of all members after adding the new member.
    */
   public List<Member> getMembers() {
+    List<Member> members = membersRef.get();
+    if (members == null) {
+      synchronized (membersRef) {
+        members = membersRef.get();
+        if (members == null) {
+          members = Util.toMembers(getResponse().getMembersList());
+          membersRef.lazySet(members);
+        }
+      }
+    }
+
```

Comment:
Yes this is what I meant, it could be as simple as:

```java 
public synchronized List<Member> getMembers() {
    if (this.members == null) {
        this.memebrs = Util.toMembers(getResponse().getMembersList());
    }

    return this.members;
}
```